src/app/services/favourites.service.ts

```diff
@@ -0,0 +1,124 @@
+import { Injectable, inject } from '@angular/core';
+import { CollectionReference, Firestore, addDoc, deleteDoc, doc, getDoc, getDocs, query, setDoc, where } from '@angular/fire/firestore';
+import { collection } from '@angular/fire/firestore';
+import { User } from '@angular/fire/auth';
+import { AuthenticationService } from './authentication.service';
+import { Activity } from '../model/activity';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class FavouritesService {
+  private firestore: Firestore = inject(Firestore);
+  private currentUser: User | null;
+  
+  favouritesCollection: CollectionReference;
+
+  constructor(private authService: AuthenticationService) {
+    this.setupAuthStatusListener();
+  }
+
+  private setupAuthStatusListener() {
+    // Subscribe to the authStatusSub BehaviorSubject in the AuthenticationService
+    this.authService.currentAuthStatus.subscribe((user) => {
+      this.currentUser = user;
+      // Do something with the user or perform actions when the authentication state changes
+      // Update favouritesCollection when the authentication state changes
+      const userUid = this.currentUser?.uid;
+
+      this.favouritesCollection = userUid
+      ? collection(this.firestore, 'users', userUid, 'favourites')
+      : (null as unknown as CollectionReference);
+    });
+  }
+
+  async isActivityFavorited(activityId: string | null): Promise<boolean> {
```

Comment:
Can we have a local in-memory array of favourites instead of querying db for every activity?

Reply:
I've removed the activity query, we can just check in the users array now